Rewrite of prevent plant model, initial commit

ash-gro opened the pull request:
This is new code for prevent plant model which combines corn and soy (underlying model was not changed). Separate files for corn/soy models are deleted.

ash-gro committed:
Rewrite of prevent plant model, initial commit

nemozen committed:
improve intro text and style, add links to dictionary

nemozen commented:
> View / edit / reply to [this conversation](https://app.reviewnb.com/gro-intelligence/api-client/pull/111/discussion/) on ReviewNB ([backstory](https://github.com/ReviewNB/support/issues/29) for this conversation format).
> 
> _tolothy commented on 2019-10-02T20:14:12Z_
> ----------------------------------------------------------------
> 
> I think line 58 doesnt need API access to look up the region_id? which means it could still fail later with authentication but still succeed here.
> 
> 
> _ash-gro commented on 2019-10-02T21:03:49Z_
> ----------------------------------------------------------------
> 
> Yes. Lookup function does not require authentication (not sure why), so this only tests that one has connection to working server but not necessarily that user has access...

static ontology info is supposed to be accessible lookup is the api client equivalent of public dictionary pages e.g. https://app.gro-intelligence.com/dictionary/regions/100000101

nemozen committed:
more text cleanup and links to ontology

nemozen committed:
expose logger via get_logger()

nemozen committed:
simplify README

nemozen commented:
> View / edit / reply to [this conversation](https://app.reviewnb.com/gro-intelligence/api-client/pull/111/discussion/) on ReviewNB ([backstory](https://github.com/ReviewNB/support/issues/29) for this conversation format).
> 
> _iba3 commented on 2019-10-04T18:40:35Z_
> ----------------------------------------------------------------
> 
> If we're no longer using get_data_series, perhaps we can remove all mention of it since it adds a substantial amount of overhead and is no longer relevant. In addition, maybe we can mention that one can use the Gro App to find out the Gro entity parameters of the data they seek.

actually in get_yearly_data() it is still using get_data_series() followed by get_data_points for the series it found.  and that is the preferred style here. we don't want the user to have to copy paste api snippets for hundreds of series. 

but a the  get_data_points is duplicating some of the functionality of client.get_df()
we should be able to refactor it so it's usign the client lib fn as a separate improvement

nemozen committed:
make plot() backward compatible with py 2.7, more text cleanup, improved logging

nemozen committed:
Merge remote-tracking branch 'origin/development' into prevent_plant_rewrite

nemozen committed:
Merge remote-tracking branch 'origin/development' into prevent_plant_rewrite

nemozen committed:
whitespace

nemozen commented:
LGTM. Needs one more round of improvements to take better advantage of some GroClient features. But this is an improvement

ash-gro committed:
Taking care of PR comments

ash-gro commented:
I committed all changes addressing comments above. I believe Nemo already approved this branch for merging into development, so unless changes in the most recent commit have to be formally checked/re-approved, someone should go ahead and do the merge (I do not have the power to do this).

tolothy reviewed (approved):
looks good to me now

JGrinst reviewed (changes requested):
The model failed for me in the 4th code block with this error:
![image](https://user-images.githubusercontent.com/45766323/68425559-57da9b80-0174-11ea-9c1a-247808600e9e.png)

nemozen committed:
refactor add_data_series to use a separate find_data_series function

no change in functionality, just separating the part that affects stored state

nemozen committed:
add time range before ranking in find_data_series

Default behavior remains the same (i.e. time range is the max
available). But if there's more than one series available that
satisfies the selection, then the time range can affect optimal
ranking so we should pass it in.

nemozen committed:
Merge branch 'prevent_plant_rewrite' of https://github.com/gro-intelligence/api-client into prevent_plant_rewrite

ash-gro commented:
get_logger is a new method of GroClient which was added (by Nemo) within
this branch. Please make sure you are pulling all the files, not just those
in samples/similar_regions and running the client which was pulled with the
rest of files (as opposed to installed with pip or in some other way)

On Thu, Nov 7, 2019 at 3:36 PM JGrinst <notifications@github.com> wrote:

> *@JGrinst* requested changes on this pull request.
>
> The model failed for me in the 4th code block with this error:
> [image: image]
> <https://user-images.githubusercontent.com/45766323/68425559-57da9b80-0174-11ea-9c1a-247808600e9e.png>
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/gro-intelligence/api-client/pull/111?email_source=notifications&email_token=ANJP3IFOGRVW43IW5Z6RVYTQSR35BA5CNFSM4I3JOD6KYY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOCKYXNYY#pullrequestreview-313620195>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ANJP3ICET5LUL5H2TJIEIQLQSR35BANCNFSM4I3JOD6A>
> .
>

nemozen committed:
minor cleanup: use logger for log messages and only use print() for output

nemozen committed:
rerun with crop=soy for complete notebook output

nemozen committed:
Merge remote-tracking branch 'origin/development' into prevent_plant_rewrite

nemozen commented:
> get_logger is a new method of GroClient which was added (by Nemo) within this branch. Please make sure you are pulling all the files, not just those in samples/similar_regions and running the client which was pulled with the rest of files (as opposed to installed with pip or in some other way)
> […](#)
> On Thu, Nov 7, 2019 at 3:36 PM JGrinst ***@***.***> wrote: *@JGrinst* requested changes on this pull request. The model failed for me in the 4th code block with this error: [image: image] <https://user-images.githubusercontent.com/45766323/68425559-57da9b80-0174-11ea-9c1a-247808600e9e.png> — You are receiving this because you authored the thread. Reply to this email directly, view it on GitHub <#111?email_source=notifications&email_token=ANJP3IFOGRVW43IW5Z6RVYTQSR35BA5CNFSM4I3JOD6KYY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOCKYXNYY#pullrequestreview-313620195>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/ANJP3ICET5LUL5H2TJIEIQLQSR35BANCNFSM4I3JOD6A> .

yes, you just need to git pull this branch before running the notebook

nemozen commented:
added a few more minor cleanups of text and logging. retested  with crop=soy as well. looks good!
